packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1068,9 +1134,42 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this._summarizer;
     }
 
-    private get summariesDisabled(): boolean {
-        return this.runtimeOptions.summaryOptions.disableSummaries === true ||
-            this.runtimeOptions.summaryOptions.summaryConfigOverrides?.disableSummaries === true;
+    public get summariesDisabled(): boolean {
+        if (this.runtimeOptions.summaryOptions.disableSummaries === true) {
+            return true;
+        }
+        return this.summaryConfiguration.state === "disabled";
+    }
+
+    public get heuristicsDisabled(): boolean {
```

Comment:
These getter methods are public are added just for unit testing? Do you think there is value in keeping them private and finding other way to test?

Reply:
Not sure - it might be useful for consumers as well. Don't you think ?

Reply:
I'd try to make them private, and test code using settings that it supplies - not much value in testing that we remember single property correctly :) Unless there is some back-comat requirements.

Reply:
Same note RE back compat comment

Reply:
Having on the test side will allow us that we have the right property value being set when we run.  If not,  we can't test whether the override is working as expected.

Reply:
I will remove from test.